### PR TITLE
adding user show page

### PR DESCRIPTION
**Pull request by johndowd**
-added migrations for the username column in users
-added model schema for username and validations
-added username input on user registration and error validation
-added new client routes for /users/:id
-added server routes for get /users/:id
-added to my profile button in Top Bar

### CODE REVIEW

File: client/src/components/layout/TopBar.js [hunk 1]
```diff
@@ -14,7 +14,18 @@ const TopBar = ({ user }) => {
     </li>,
   ];
 
+  let id = ""
+  if(user){
+    id = user.id
+  }
+
+
```
Comment (dgazzola) [inline client/src/components/layout/TopBar.js]: extra empty line here

File: server/src/models/User.js [hunk 2]
```diff
@@ -26,9 +26,10 @@ class User extends uniqueFunc(Model) {
   static get jsonSchema() {
     return {
       type: "object",
-      required: ["email"],
+      required: ["email", "username"],
```
Comment (dgazzola) [inline server/src/models/User.js]: empty line

**Review by dgazzola — COMMENTED** [review]
Nice this looks all set just need to remove 1 or 2 extra empty lines (think I commented) all good after that!

File: client/src/components/UserShowPage.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    const getUser = async () => {
+        try {
+            const response = await fetch(`/api/v1/users/${id}`)
+            const body = await response.json()
+            setUser(body.user)
+        } catch (error) {
+            console.error(error)
+        }
+    }
```
Comment (nwalberts) [inline client/src/components/UserShowPage.js:21]: lets use slightly better error handling please

File: client/src/components/layout/TopBar.js [hunk 4]
```diff
@@ -14,7 +14,17 @@ const TopBar = ({ user }) => {
     </li>,
   ];
 
+  let id = ""
+  if(user){
+    id = user.id
+  }
```
Comment (nwalberts) [inline client/src/components/layout/TopBar.js:20]: this is okay, but in the future use optional chaining
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

File: client/src/components/registration/RegistrationForm.js [hunk 5]
```diff
@@ -96,6 +96,17 @@ const RegistrationForm = () => {
...
           </label>
         </div>
+        <div>
+          <label>
+            User Name:
+            <input 
+              type="text" 
+              name="username" 
+              value={userPayload.username} 
+              onChange={onInputChange}/>
```
Comment (nwalberts) [inline client/src/components/registration/RegistrationForm.js]: indentation is off. the `/> should be on a newline and inline with the `<input`

File: server/src/models/User.js [hunk 6]
```diff
@@ -26,9 +26,9 @@ class User extends uniqueFunc(Model) {
   static get jsonSchema() {
     return {
       type: "object",
-      required: ["email"],
-
+      required: ["email", "username"],
       properties: {
+        username: { type: "string", minLength: 3, maxLength: 15 },
```
Comment (nwalberts) [inline server/src/models/User.js:31]: you should be adding `username` to the unique factory above as well!

File: server/src/routes/api/v1/usersRouter.js [hunk 7]
```diff
@@ -4,10 +4,21 @@ import { User } from "../../../models/index.js";
 
 const usersRouter = new express.Router();
 
+usersRouter.get("/:id", async (req, res) => {
+  const { id } = req.params
+  console.log(id)
```
Comment (nwalberts) [inline server/src/routes/api/v1/usersRouter.js]: no console.logs allowed

**Review by nwalberts — CHANGES_REQUESTED** [review]

File: client/src/components/layout/TopBar.js (hunk 4 above)
Comment (johndowd) [reply client/src/components/layout/TopBar.js:20]: i couldn't figure out the option changing, ill look into more later

**Pull request merged**